### PR TITLE
feat(origin-core-ui): improvements to user profile page

### PR DESCRIPTION
**Pull request by marcin-l-tsh**
#### before:
screencapture-localhost-3000-account-user-profile-2021-01-04-13_52_04
#### after:
screencapture-localhost-3000-account-user-profile-2021-01-04-13_42_48

### CODE REVIEW

File: packages/origin-ui-core/src/components/Account/UserProfile/ChangePassword.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+                            {isEditing && (
+                                <Button
+                                    style={{ marginRight: 10 }}
+                                    type="button"
+                                    variant="contained"
+                                    color="primary"
+                                    className="mt-3 right"
+                                    disabled={!(touched.currentPassword || touched.newPassword)}
+                                    onClick={async () => {
```
Comment (alexworker23) [inline packages/origin-ui-core/src/components/Account/UserProfile/ChangePassword.tsx:130]: When clicking Save in ChangePassword the fields don't get empty. Save and Cancel buttons also don't hide.

File: packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            display: 'flex',
+            alignItems: 'center'
+        },
+        infoIcon: {
+            marginLeft: '1rem'
+        }
+    })
+);
+
+export const useUserProfileStyle = () => useStyles(useTheme());
```
Comment (alexworker23) [inline packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx]: Why do you need UserProfile styles as a separate component? Why don't include it into the `UserDataEditForm.tsx` ?

File: packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                                </Grid>
+                            </Grid>
+                            {isEditing && (
+                                <Button
+                                    style={{ marginRight: 10 }}
+                                    type="button"
+                                    variant="contained"
+                                    color="primary"
+                                    className="mt-3 right"
+                                    disabled={
```
Comment (alexworker23) [inline packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx:210]: After first change of data Save button is no longer disabled.

File: packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                                    onClick={async () => {
+                                        await formikProps.validateForm();
+                                        await formikProps.submitForm();
+                                        setIsEditing(false);
+                                    }}
+                                >
+                                    {t('user.actions.save')}
+                                </Button>
+                            )}
+                            {isEditing && (
```
Comment (alexworker23) [inline packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx:225]: Same here. If you change some data and then click Cancel - your Save and Cancel buttons will no longer be disabled by default.

File: packages/origin-ui-core/src/components/Account/UserProfile/UserEmailChange.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                                    {t('user.actions.save')}
+                                </Button>
+                            )}
+                            {isEditing && (
+                                <Button
+                                    type="button"
+                                    variant="contained"
+                                    color="primary"
+                                    className="mt-3 right"
+                                    onClick={() => {
```
Comment (alexworker23) [inline packages/origin-ui-core/src/components/Account/UserProfile/UserEmailChange.tsx:122]: When changing email and then clicking Cancel - Save button is no longer disabled.

**Review by alexworker23 — CHANGES_REQUESTED** [review]

**Comment by josipbagaric** [comment]
@marcin-l-tsh Does something change on the user interface?
If so, could you attach screenshots to the PR? (Ideally a before/after comparison).

We do this for UI PRs to get a better sense of what's changed and how to review the PR.

File: packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx (hunk 2 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx]: it's also used in every other part of user profile page (ChangePassword, UserEmailChange, etc). should I use different approach is case like this?

Comment (alexworker23) [reply packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx]: Then you can create an scss file, like UserProfile.scss in styles folder and apply classes

File: packages/origin-ui-core/src/components/Account/UserProfile/ChangePassword.tsx (hunk 1 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/ChangePassword.tsx:130]: fixed, now after fields should be cleared after saving or canceling

File: packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx (hunk 3 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx:210]: fixed
saving with success or canceling should now quit editing mod, then re-entering should keep save button disabled initially.

File: packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx (hunk 4 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/UserDataEditForm.tsx:225]: fixed

File: packages/origin-ui-core/src/components/Account/UserProfile/UserEmailChange.tsx (hunk 5 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/UserEmailChange.tsx:122]: fixed, now should be disabled after re-entering editing mode

**Comment by marcin-l-tsh** [comment]
> @marcin-l-tsh Does something change on the user interface?
> If so, could you attach screenshots to the PR? (Ideally a before/after comparison).
> 
> We do this for UI PRs to get a better sense of what's changed and how to review the PR.

screenshots added

File: packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx (hunk 2 above)
Comment (marcin-l-tsh) [reply packages/origin-ui-core/src/components/Account/UserProfile/Styles.tsx]: done, styles moved to scss file

**Review by alexworker23 — COMMENTED** [review]
When user changes e-mail or password he is logged out, but the url stays the same `/account/user-profile`, which is causing the empty page to appear.
The solution is to configure redirect to the default page.

**Review by alexworker23 — APPROVED** [review]

**Pull request merged**